File: app/src/main/java/com/sd/views/MaxSizeLayoutActivity.java
package com.sd.views;

import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

import com.sd.views.databinding.ActivityMaxSizeLayoutBinding;

public class MaxSizeLayoutActivity extends AppCompatActivity {
    public static final String TAG = MaxSizeLayoutActivity.class.getSimpleName();

    private ActivityMaxSizeLayoutBinding mBinding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mBinding = ActivityMaxSizeLayoutBinding.inflate(getLayoutInflater());
        setContentView(mBinding.getRoot());

        // 设置最大宽度
        mBinding.maxSizeLayout.setMaxWidth(300);
        // 设置最大高度
        mBinding.maxSizeLayout.setMaxHeight(300);

        mBinding.btnSmall.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mBinding.tvContent.setText("123456789");
            }
        });
        mBinding.btnBig.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                final String content = mBinding.tvContent.getText().toString() + mBinding.tvContent.getText().toString();
                mBinding.tvContent.setText(content);
            }
        });
    }
}

File: app/src/main/java/com/sd/views/MaxSizeLayoutActivity.kt
package com.sd.views

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.sd.views.databinding.ActivityMaxSizeLayoutBinding

class MaxSizeLayoutActivity : AppCompatActivity(), View.OnClickListener {
    private lateinit var _binding: ActivityMaxSizeLayoutBinding
    private val _content = "123456789"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        _binding = ActivityMaxSizeLayoutBinding.inflate(layoutInflater)
        setContentView(_binding.root)

        // 设置最大宽度
        _binding.maxSizeLayout.setMaxWidth(400)
        // 设置最大高度
        _binding.maxSizeLayout.setMaxHeight(400)
    }

    override fun onClick(v: View?) {
        when (v) {
            _binding.btnSmall -> _binding.tvContent.text = _content
            _binding.btnBig -> {
                val text = _binding.tvContent.text.toString()
                _binding.tvContent.text = text + _content
            }
        }
    }
}